### ISSUE TITLE
IllegalArgumentException when try to upload a file with multipart/form-data & http-builder-ng-okhttp

### ISSUE THREAD

**Issue by vasogiamarelou** [http-builder-ng/http-builder-ng#219]
Hello,

we are trying to upload a file(.png) with http-builder-ng-okhttp 
the code is the following 

```
import static groovyx.net.http.MultipartContent.multipart

            def builder = HttpBuilder.configure {
            request.uri = 'myUrl'
            request.contentType = 'multipart/form-data'
        }

        builder.post {
            request.uri.path = '/path'
            request.contentType = 'multipart/form-data'
            request.encoder'multipart/form-data' , CoreEncoders.&multipart
            request.body = multipart {
                part 'image', 'myImage.png', 'multipart/form-data', new File('myImage.png')
            }
        }

```
and we get the following error

```
java.lang.IllegalArgumentException: Multipart body content must be MultipartContent.

	at groovyx.net.http.CoreEncoders.multipart(CoreEncoders.java:50)
	at groovy.lang.Closure.call(Closure.java:411)
	at groovyx.net.http.EmbeddedEncoder.encode(EmbeddedEncoder.java:37)
	at groovyx.net.http.CoreEncoders$EncodedDataSource.getInputStream(CoreEncoders.java:108)
	at javax.activation.DataHandler.writeTo(DataHandler.java:305)
	at javax.mail.internet.MimeBodyPart.writeTo(MimeBodyPart.java:452)
	at javax.mail.internet.MimeMultipart.writeTo(MimeMultipart.java:157)
	at groovyx.net.http.CoreEncoders.multipart(CoreEncoders.java:67)
	at groovy.lang.Closure.call(Closure.java:411)
	at groovyx.net.http.JavaHttpBuilder$Action.lambda$execute$2(JavaHttpBuilder.java:138)
	at groovyx.net.http.JavaHttpBuilder$ThreadLocalAuth.with(JavaHttpBuilder.java:331)
	at groovyx.net.http.JavaHttpBuilder$Action.execute(JavaHttpBuilder.java:122)
	at groovyx.net.http.JavaHttpBuilder.createAndExecute(JavaHttpBuilder.java:374)
	at groovyx.net.http.JavaHttpBuilder.doPost(JavaHttpBuilder.java:389)
	at groovyx.net.http.HttpObjectConfigImpl.nullInterceptor(HttpObjectConfigImpl.java:47)
	at groovyx.net.http.HttpBuilder.post(HttpBuilder.java:732)
	at groovyx.net.http.HttpBuilder.post(HttpBuilder.java:1391)
```

Do you have any idea what is missing from our code or is it something else?
please inform me if you want any further details/logs 

Thank you

**Comment by cjstehno** [comment]
The encoder is client-specific. Please try `OkHttpEncoders.&multipart` rather than using the one from `CoreEncoders` (see also: [User Guide: Multipart](https://http-builder-ng.github.io/http-builder-ng/asciidoc/html5/#_multipart)).

If this does not fix the issue, please provide a failing test so that it may be reproduced, preferably using [Ersatz](http://stehno.com/ersatz/).

**Comment by vasogiamarelou** [comment]
Thank you a lot @cjstehno. It works.
one more thing that I notice is that in my first try for the `contentType` inside the part block I used `multipart/form-data` but when I changed it to `text/plain` as it is in the user guide it worked again. But i am not sure that i get the difference. I expected to be `multipart/form-data` everywhere.

P.S  we just started using http-builder-ng into our project so more questions may follow.
Thank you a lot for your support and help.
